### PR TITLE
updated to refractor code

### PR DESCRIPTION
**Pull request by kingqabil**
Refactored code as per SOLID principle.
Give single responsibility to main.rb to run the app and handle user inputs.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -1,162 +1,29 @@
...
+      actions
+      option = gets.chomp
+      break if option == '7'
+      @app.handle_action option
+    end
+    puts 'Thank you for using this app!'
+  end
+end
+run_application = Main.new
+run_application.run
```
Comment (rloterh) [inline main.rb]: - [ ] The role of _main.rb_ is only to start their code and ask the user for the initial choice of options (or to quit). It should not have extra methods. You refactor your _main.rb_ in this manner:
```ruby
require "./app.rb"

def main
  app = App.new()

  response = nil

  puts "Welcome to School Library App!\n\n"
  while response != "7"
    puts "Please choose an option by enterin a number:"
    puts "1 - List all books"
    puts "2 - List all people"
    puts "3 - Create a person"
    puts "4 - Create a book"
    puts "5 - Create a rental"
    puts "6 - List all rentals for a given person id"
    puts "7 - Exit"
    response = gets.chomp

    case response
    when "1"
      app.list_books()
    when "2"
      app.list_people()
    when "3"
      app.create_person()
    when "4"
      app.create_book()
    when "5"
      app.create_rental()
    when "6"
      app.list_rentals_for_person_id()
    when "7"
      puts "Thank you for using this app!"
    end

    puts "\n"
  end
end

main()
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Dire  & Sadiq, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- Descriptive Pull Request
- Good ReadMe
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb [hunk 2]
```diff
@@ -1,162 +1,42 @@
...
+    when '2'
+      app.list_people
+    when '3'
+      app.create_person
+    when '4'
+      app.create_book
+    when '5'
+      app.create_rental
+    when '6'
+      app.list_rentals_for_person_id
```
Comment (aliabbani) [inline main.rb]: I think we need to change `app.list_rentals_for_person_id` to `app.list_rentals` .
Please make sure the app works correctly in the console.

**Review by aliabbani — CHANGES_REQUESTED** [review]
Hi @kingqabil ✋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by o-rashad — APPROVED** [review]
Asslam Alaikum Team @kingqabil @direakanbi  ,😊

Hi 👋🏽, 
`Thanks for implementing the changes required by the previous code revisers`
Your project is complete!  
There is nothing else to say other than... it's time to merge it 😎 

# Congratulations! 🎉 

Happy coding!👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**